### ISSUE TITLE
Continual learning through synaptic intelligence

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#25]
## [Continual learning through synaptic intelligence](https://dl.acm.org/doi/10.5555/3305890.3306093)
*While deep learning has led to remarkable advances across diverse applications, it struggles in domains where the data distribution changes over the course of learning. In stark contrast, biological neural networks continually adapt to changing domains, possibly by leveraging complex molecular machinery to solve many tasks simultaneously. In this study, we introduce intelligent synapses that bring some of this biological complexity into artificial neural networks. Each synapse accumulates task relevant information over time, and exploits this information to rapidly store new memories without forgetting old ones. We evaluate our approach on continual learning of classification tasks, and show that it dramatically reduces forgetting while maintaining computational efficiency.*

### Cite this paper
```bibtex
@inproceedings{10.5555/3305890.3306093,
    author = {Zenke, Friedemann and Poole, Ben and Ganguli, Surya},
    title = {Continual Learning through Synaptic Intelligence},
    year = {2017},
    publisher = {JMLR.org},
    booktitle = {Proceedings of the 34th International Conference on Machine Learning - Volume 70},
    pages = {3987–3995},
    numpages = {9},
    location = {Sydney, NSW, Australia},
    series = {ICML’17}
}
  

```

**Comment by juanmc2005** [comment]
### Description

This paper proposes a method to tackle catastrophic forgetting, that is very similar to EWC, with the difference that weight importance is calculated in an online fashion during training using the already available gradients. In contrast, EWC needs to approximate offline the diagonal of the Fisher information matrix using the last task's parameters.

### Method

It is a regularization method consisting of a surrogate loss that is added to the main loss (cross entropy in their experiments).

### Experiments

They experiment on sequential versions of MNIST, CIFAR-10 and CIFAR-100 with two different architectures.

### Results

They test their method with and without consolidation, as well as EWC and a control baseline.
Overall, the method performs equally well as EWC, with the advantage of online importance calculation.

### Takeaway

No immediate takeaways, other than the fact that this seems to be a suitable replacement for EWC to reduce computational complexity if we ever want to implement it.
